### ISSUE TITLE
[IOSXE] Parse  'show monitor session all'  fail

### ISSUE THREAD

**Issue by hsang88** [CiscoTestAutomation/genieparser#882]
###  config local erspan on a IOSXE device

```
charon4p_p1b#conf terminal
Enter configuration commands, one per line.  End with CNTL/Z.
charon4p_p1b(config)#monitor session 10 type erspan-source
charon4p_p1b(config-mon-erspan-src)# no shutdown
charon4p_p1b(config-mon-erspan-src)# source interface Gi0/0/1
charon4p_p1b(config-mon-erspan-src)# destination
charon4p_p1b(config-mon-erspan-src-dst)#  erspan-id 10
charon4p_p1b(config-mon-erspan-src-dst)#  mtu 1464
charon4p_p1b(config-mon-erspan-src-dst)#  ip address 10.1.1.1
charon4p_p1b(config-mon-erspan-src-dst)#  origin ip address 100.1.1.1
charon4p_p1b(config-mon-erspan-src-dst)#monitor session 20 type erspan-destination
charon4p_p1b(config-mon-erspan-dst)# no shutdown
charon4p_p1b(config-mon-erspan-dst)# destination interface Gi0/1/3
charon4p_p1b(config-mon-erspan-dst)# source
charon4p_p1b(config-mon-erspan-dst-src)#  erspan-id 10
charon4p_p1b(config-mon-erspan-dst-src)#  ip address 10.1.1.1
charon4p_p1b(config-mon-erspan-dst-src)#end
charon4p_p1b#
```

###  parse 'show monitor session 10' OK

```
parsed = dev.parse('show monitor session 10')

2024-08-26 07:49:16,779: %UNICON-INFO: +++ charon4p_p1b with via 'cli': executing command 'show monitor session 10' +++
show monitor session 10
Session 10
----------
Type                     : ERSPAN Source Session
Status                   : Admin Enabled
Source Ports             : 
    Both                 : Gi0/0/1
Destination IP Address   : 10.1.1.1
MTU                      : 1464
Destination ERSPAN ID    : 10
Origin IP Address        : 100.1.1.1


charon4p_p1b#
>>> pprint(parsed)
{'session': {'10': {'destination_erspan_id': '10',
                    'destination_ip_address': '10.1.1.1',
                    'mtu': 1464,
                    'origin_ip_address': '100.1.1.1',
                    'source_ports': {'both': 'Gi0/0/1'},
                    'status': 'Admin Enabled',
                    'type': 'ERSPAN Source Session'}}}
```

###  but parse 'show monitor session all' fail

```
>>> parsed = dev.parse('show monitor session all')

2024-08-26 07:49:35,657: %UNICON-INFO: +++ charon4p_p1b with via 'cli': executing command 'show monitor' +++
show monitor
show monitor
% Incomplete command.

charon4p_p1b#
Traceback (most recent call last):
  File "/nobackup/yuhe3/workspace/hsang/workspace/polaris/binos/linkfarm/sdks-host/iosxe-pyats/nativesdk/sdk/sysroots/x86_64-xesdk-linux/usr/lib/python3.10/site-packages/unicon/plugins/generic/service_implementation.py", line 741, in call_service
    self.result = self.get_service_result()
  File "/nobackup/yuhe3/workspace/hsang/workspace/polaris/binos/linkfarm/sdks-host/iosxe-pyats/nativesdk/sdk/sysroots/x86_64-xesdk-linux/usr/lib/python3.10/site-packages/unicon/bases/routers/services.py", line 235, in get_service_result
    raise SubCommandFailure(
unicon.core.errors.SubCommandFailure: ('sub_command failure, patterns matched in the output:', ['^%\\s*[Ii]ncomplete (command|input)'], 'service result', 'show monitor\r\nshow monitor\r\n% Incomplete command.\r\n\r\ncharon4p_p1b#')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/nobackup/yuhe3/workspace/hsang/workspace/polaris/binos/linkfarm/sdks-host/iosxe-pyats/nativesdk/sdk/sysroots/x86_64-xesdk-linux/usr/lib/python3.10/site-packages/genie/metaparser/_metaparser.py", line 308, in parse
    output = call(**kwargs)
  File "/nobackup/yuhe3/workspace/hsang/workspace/polaris/binos/linkfarm/sdks-host/iosxe-pyats/nativesdk/sdk/sysroots/x86_64-xesdk-linux/usr/lib/python3.10/site-packages/genie/libs/parser/iosxe/show_monitor.py", line 84, in cli
    out = self.device.execute(cmd)
  File "/nobackup/yuhe3/workspace/hsang/workspace/polaris/binos/linkfarm/sdks-host/iosxe-pyats/nativesdk/sdk/sysroots/x86_64-xesdk-linux/usr/lib/python3.10/site-packages/unicon/bases/routers/services.py", line 259, in __call__
    self.call_service(*args, **kwargs)
  File "/nobackup/yuhe3/workspace/hsang/workspace/polaris/binos/linkfarm/sdks-host/iosxe-pyats/nativesdk/sdk/sysroots/x86_64-xesdk-linux/usr/lib/python3.10/site-packages/unicon/plugins/generic/service_implementation.py", line 746, in call_service
    raise SubCommandFailure("Command execution failed", err) from err
unicon.core.errors.SubCommandFailure: ('Command execution failed', SubCommandFailure('sub_command failure, patterns matched in the output:', ['^%\\s*[Ii]ncomplete (command|input)'], 'service result', 'show monitor\r\nshow monitor\r\n% Incomplete command.\r\n\r\ncharon4p_p1b#'))

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/nobackup/yuhe3/workspace/hsang/workspace/polaris/binos/linkfarm/sdks-host/iosxe-pyats/nativesdk/sdk/sysroots/x86_64-xesdk-linux/usr/lib/python3.10/site-packages/genie/conf/base/device.py", line 535, in parse
    return self._get_parser_output(kwargs, parsers[0], parsers[1],
  File "/nobackup/yuhe3/workspace/hsang/workspace/polaris/binos/linkfarm/sdks-host/iosxe-pyats/nativesdk/sdk/sysroots/x86_64-xesdk-linux/usr/lib/python3.10/site-packages/genie/conf/base/device.py", line 572, in _get_parser_output
    out = parser_instance.parse(**copy)
  File "/nobackup/yuhe3/workspace/hsang/workspace/polaris/binos/linkfarm/sdks-host/iosxe-pyats/nativesdk/sdk/sysroots/x86_64-xesdk-linux/usr/lib/python3.10/site-packages/genie/metaparser/_metaparser.py", line 311, in parse
    raise InvalidCommandError from e
genie.metaparser.util.exceptions.InvalidCommandError: Invalid command has been executed

```

**Comment by SohanTirpude** [comment]
Hello @hsang88,

Currently I am looking into this. Can you please share the raw device output for `show monitor session all` command?

Thank you.

**Comment by hsang88** [comment]
> Hello @hsang88,
> 
> Currently I am looking into this. Can you please share the raw device output for `show monitor session all` command?
> 
> Thank you.

charon4p_p1b#conf t
Enter configuration commands, one per line.  End with CNTL/Z.
charon4p_p1b(config)#monitor session 10 type erspan-source
charon4p_p1b(config-mon-erspan-src)# no shutdown
charon4p_p1b(config-mon-erspan-src)# source interface Gi0/0/1
charon4p_p1b(config-mon-erspan-src)# destination
charon4p_p1b(config-mon-erspan-src-dst)#  erspan-id 10
charon4p_p1b(config-mon-erspan-src-dst)#  mtu 1464
charon4p_p1b(config-mon-erspan-src-dst)#  ip address 10.1.1.1
charon4p_p1b(config-mon-erspan-src-dst)#  origin ip address 100.1.1.1
charon4p_p1b(config-mon-erspan-src-dst)#monitor session 20 type erspan-destination
charon4p_p1b(config-mon-erspan-dst)# no shutdown
charon4p_p1b(config-mon-erspan-dst)# destination interface Gi0/1/3
charon4p_p1b(config-mon-erspan-dst)# source
charon4p_p1b(config-mon-erspan-dst-src)#  erspan-id 10
charon4p_p1b(config-mon-erspan-dst-src)#  ip address 10.1.1.1
charon4p_p1b(config-mon-erspan-dst-src)#
charon4p_p1b(config-mon-erspan-dst-src)#end
charon4p_p1b#
charon4p_p1b#show monitor session all
Session 10
----------
Type                     : ERSPAN Source Session
Status                   : Admin Enabled
Source Ports             :
    Both                 : Gi0/0/1
Destination IP Address   : 10.1.1.1
MTU                      : 1464
Destination ERSPAN ID    : 10
Origin IP Address        : 100.1.1.1


Session 20
----------
Type                     : ERSPAN Destination Session
Status                   : Admin Enabled
Destination Ports        : Gi0/1/3
Source IP Address        : 10.1.1.1
Source ERSPAN ID         : 10


charon4p_p1b#

**Comment by SohanTirpude** [comment]
Hello @hsang88 ,

The PR got merged and the fix will be available in the upcoming pyats 24.9 version. Kindly try and let me know if you still see this issue with pyats 24.9 version.

Meanwhile, I am closing this ticket. If you are facing any other issues, kindly open a new ticket.

Thank you.
